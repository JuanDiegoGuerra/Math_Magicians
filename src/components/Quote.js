import { useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState({});
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchQuote = async () => {
      setLoading(true);
      try {
        const res = await fetch(
          `https://api.api-ninjas.com/v1/quotes?category=${'failure'}`,
          {
            headers: {
              'X-Api-Key': 'rUuUFv1qUBQHFwLXZP09+Q==fdgniVquw8YwHAll',
            },
          },
        );
        const json = await res.json();
        setQuote(json[0]);
      } catch (error) {
        setError(true);
      }
      setLoading(false);
    };
    fetchQuote();
  }, []);

  if (error) {
    return (
      <p className="error-on-fetch">Something went wrong!</p>
    );
  }
  if (loading) {
    return (
      <p className="loading-quote">Loading...</p>
    );
  }

  return (
    <p className="quote">
      <i>
        &quot;
        {`${quote.quote}`}
        &quot;
      </i>
      {' '}
      -
      <span className="author">{`${quote.author}`}</span>
    </p>
  );
};

export default Quote;
