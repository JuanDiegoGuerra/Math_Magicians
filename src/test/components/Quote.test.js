import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Quote from '../../components/Quote';

describe('Quote', () => {
  test('Renders correctly', () => {
    const { container } = render(<Quote />);
    expect(container).toMatchSnapshot();
  });

  test('Displays loading message while fetching quotes', () => {
    const { getByText } = render(<Quote />);
    const loading = getByText('Loading...');
    expect(loading).toBeInTheDocument();
  });
});