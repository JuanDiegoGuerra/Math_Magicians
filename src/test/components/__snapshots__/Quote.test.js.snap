// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Renders correctly 1`] = `
<div>
  <p
    class="loading-quote"
  >
    Loading...
  </p>
</div>
`;
